### PR TITLE
Initial data fetch for Most Watched Page

### LINKED ISSUE #7641: Fetch data for most watched page & render most watched component
Example map: [Miro](https://miro.com/app/board/o9J_kvZ2wMk=/?moveToWidget=3074457349544789098&cot=12)
Some analysis of existing routes: [Dropbox Paper](https://paper.dropbox.com/doc/Analysis-Most-Watched--A6ccw62pyGe4I0pgfBjDw8iDAg-TK3nz2VKkDvUNC8P1rnh7) 
Jira ticket: [Jira](https://jira.dev.bbc.co.uk/browse/WSMEDIA-234)
Design: [Zeplin](https://app.zeplin.io/project/5d4d77b1fcbfc79bf47e0294/screen/5f22de67d3d1fd1f74e4cda1)

Blocked until the basic page routing is done in https://github.com/bbc/simorgh/issues/7640

**Is your feature request related to a problem? Please describe.**
The user should be able to view the most watched component when visiting e.g. https://www.bbc.com/afaanoromoo/media/video

**Describe the solution you'd like**

- add `getInitialData` function in route module to fetch most watched data and pass to page component (route must have been set up in https://github.com/bbc/simorgh/issues/7640)

Add Most watched component to page
 - https://github.com/bbc/simorgh/blob/latest/src/app/containers/MostWatched/index.jsx
- The Most watched component requires some changes
  - should not display the h2 most watched title
  - add logic so the promo headline is an h2 on most watched pages (not an h3 as it currently is when not on most watched page)

Question: How many items do we show? Currently in PAL there are 20

**A11y Acceptance Criteria**
Most Watched should be within the main container and the h1 should be visually displayed as ‘Most Watched'

The skip to content link should anchor here (note there should not be an h2 for most watched or any other regions for most watched within the main region)

**Skip to content link**
Given I use a keyboard only
When I navigate to the banner landmark by pressing the TAB key
Then the ‘skip to content’ link must be the first piece of content to receive focus after the brand logo
And the ‘skip to content’ link must be visually displayed on focus
And this must include the browser default focus outline styling

Given I use a screen reader
When I activate the ‘skip to content’ link
Then the h1 in the main region should be announced
And after this announcement, I must be able to navigate to the next piece of content below the h1

Given there is a ‘skip to content’ link
When I activate the ‘skip to content’ link
Then the url should be user friendly and descriptive of it’s destination e.g. #content

**List of promos (within section)**
Given I use a screen reader
When I view a section with more than one promo 
And I navigate to the first promo
Then it should be announced as a list item within a list

**Promo image**
StoryPromo - Psammead unit test
https://github.com/bbc/psammead/issues/541

Given I use a screen reader
When I view a promo with an image 
And I navigate to the promo image
Then the image alt text must be announced
Promo h2

**StoryPromo - Psammead unit test**
https://github.com/bbc/psammead/issues/541

Given I use a screen reader
When I navigate to a promo
Then this must have one heading level 2
And must be announced as a link

NB: See also full promo a11y a/c: https://docs.google.com/document/d/1wC9vTLCt5RwfKIHn7spWsmwvt7INl4w1Pfrl7jD8iZA/edit#heading=h.myflp6au8u7p

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Additional context**

Blocked by https://github.com/bbc/simorgh/issues/7640

### LINKED ISSUE #7691: Control the number of items displayed on the Most Watched Pages using toggles
**Is your feature request related to a problem? Please describe.**
Editorial have requested that we ensure the number of items displayed on the most watched pages can be adjusted.
The default number to display is 10 items, but we should be able to adjust this number using iSite. 

**Describe the solution you'd like**
Ability to toggle the number of most watched items shown on a Most Watched Page. 
The number will be set to 10 by default. 

Given only 10 items are specified for the most watched page for Afrique within isite
When I visit bbc.com/afrique/media/video
then only 10 most watched items are displayed on the page

**Describe alternatives you've considered**

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Additional context**

### LINKED ISSUE #7707: Render Most Watched Component as an Ordered List
**Is your feature request related to a problem? Please describe.**
The most watched component will be rendered as an ordered list so that the list order is announced to screenreader users 

**Describe the solution you'd like**
Update ‘Most watched’ to an ‘ol’ from ‘ul’ on the MAP pages and the Most Watched page

Given I use a screen reader
When I view a section with more than one promo 
And I navigate to the first promo
Then it should be announced as a list item 1 within an ordered list

**Describe alternatives you've considered**

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by rebeccamcginn**
Resolves #7641
Resolves #7691
Resolves #7707

**Overall change:**
We fetch the Most Watched data and render the component on the Most Watched page. 

**Code changes:**

- Pass through pageData to Most Watched Page and render Most Watched component
- Added new toggle for Most Watched Page - mostWatchedMediaPage
- Added light grey bar between the Most Watched items
- Changed Most Watched component to an ordered list
- Renamed isMapContent to isMediaContent to serve both MAPs and Most Watched pages

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [x] I have assigned this PR to the Simorgh project

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [x] This PR requires manual testing

### CODE REVIEW

**Comment by rebeccamcginn** [comment]
Heading is currently incorrect 
image

File: src/app/containers/CpsOnwardJourney/index.jsx [hunk 1]
```diff
@@ -78,13 +78,17 @@ const gridMarginSmall = css`
 `;
 
 const Wrapper = styled(ConstrainedLargeGrid)`
+  ${isMediaContent =>
+    !isMediaContent &&
```
Comment (ryanmccombe) [inline src/app/containers/CpsOnwardJourney/index.jsx]: Maybe something like `Wrapper = isMapContent ? Grid : styled(Grid)` would be clearer

File: src/app/containers/CpsRelatedContent/RelatedContentPromoList/index.jsx [hunk 2]
```diff
@@ -1,11 +1,26 @@
...
+import { GEL_GROUP_4_SCREEN_WIDTH_MIN } from '@bbc/gel-foundations/breakpoints';
+import { GEL_SPACING_DBL } from '@bbc/gel-foundations/spacings';
+import { C_LUNAR } from '@bbc/psammead-styles/colours';
 import Grid from '../../../components/Grid';
 import StoryPromo from '../../StoryPromo';
 import { storyItem } from '#models/propTypes/storyItem';
 
-const RelatedContentPromoList = ({ promoItems, dir, isMapContent }) => {
+const MediaStoryPromoLi = styled(StoryPromoLi)`
+  ${({ border }) =>
```
Comment (ryanmccombe) [inline src/app/containers/CpsRelatedContent/RelatedContentPromoList/index.jsx]: where does the `border` prop come from?

**Comment by rebeccamcginn** [comment]
Heading fixed

image

**Review by ryanmccombe — APPROVED** [review]

**Review by pharingee — APPROVED** [review]
LATM 👍

**Pull request merged**